serviceregistration/kubernetes/client/client.go

```diff
@@ -0,0 +1,288 @@
+package client
+
+import (
+	"bytes"
+	"crypto/tls"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/cenkalti/backoff"
```

Comment:
This import here (and I think the package is also pulled in elsewhere) should be `github.com/cenkalti/backoff/v4`, per https://github.com/cenkalti/backoff#usage

Reply:
Also, could we use https://github.com/hashicorp/go-retryablehttp, which is already in use elsewhere in Vault, instead of bringing in a new package?

Reply:
Oh nice! I will try out using `go-retryablehttp`, and if for some reason it doesn't work, will fall back to `github.com/cenkalti/backoff/v4`. The latter is already included in our vendored stuff indirectly, FWIW.

Reply:
Hm, I did this but I'm not actually sure whether I like the code better. These are the two commits: https://github.com/hashicorp/vault/pull/8249/commits/b00a0c420174d2557d789ba1e9924aa89e73f51a, https://github.com/hashicorp/vault/pull/8249/commits/283ba7f1a443aa1e41ceb26098d9d051b76c0379.

Reply:
Marking resolved to track what I've pushed code for.

Reply:
Also, it's worth noting that using the `goretryablehttp` package has actually resulted in more lines of code so far. The client was originally in the >280 range, and now even with cutting some PatchOperation lines it's in the >290 range. I'm totally fine with either leaving it as-is now, or reverting those commits, I honestly have no attachment to either way. Just noting that it shook out to be more lines as a point of information in considering whether we'd want to standardize on this approach in the future.